internal/controllers/utils/oauth.go

```diff
@@ -11,12 +11,41 @@ import (
 
 	"golang.org/x/oauth2"
 	"golang.org/x/oauth2/clientcredentials"
+	"k8s.io/apiserver/pkg/server/dynamiccertificates"
 )
 
+// StaticKeyPairLoader provides a means by which we can use a static certificate as input into the dynamic content
+// provider interface.  We limit the use of this to clients that are a one-shot implementation that would get re-created
+// periodically and wouldn't need to worry about certificates getting renewed as part of a long lived client.
+type StaticKeyPairLoader struct {
+	cert []byte
+	key  []byte
+}
+
+// AddListener is a no-op for static providers.
+func (s *StaticKeyPairLoader) AddListener(_ dynamiccertificates.Listener) {
+	// NOOP
+}
+
+// Name is a simple name to identify this provider
+func (s *StaticKeyPairLoader) Name() string {
+	return "StaticKeyPairLoader"
+}
```

Comment:
Just curious....why have a `StaticKeyPairLoader` instead of using  `dynamiccertificates.NewDynamicServingContentFromFiles("StaticKeyPairLoader"...` (just like client-oauth) ? I guess we don't really have a path which the NewDynamicServingContentFromFiles expects?